### ISSUE TITLE
Trouble installing on Fish (Windows, MSYS2)

### ISSUE THREAD

**Issue by Terraphice** [JanDeDobbeleer/oh-my-posh#2925]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

When following the [installation docs for changing the prompt](https://ohmyposh.dev/docs/installation/prompt) for Fish I came across quite a few errors running `oh-my-posh init fish | source`.

![image](https://user-images.githubusercontent.com/41020526/194833681-20b58c3e-10a5-4b75-ba7e-87955689e0e6.png)

I'm running Windows 11 Insider, Windows Terminal Preview, and I've installed Fish through the MSYS2 installation method on the [fishshell website](https://fishshell.com/). I'm fairly confident that it being installed through MSYS2 is the problem muddying my file paths here. CDing to the '/' or 'C:/' directories didn't fair any better. Maybe it's user error, as I'm just now giving shells besides ZSH and PWSH a shot.

Expected behavior was obviously for oh-my-posh to be initialized and to acquire powerlines/a prompt. I had 0 issues installing Oh-My-Posh to PWSH, Nu, or the MSYS2 versions of bash or ZSH.

Oh-My-Posh does run otherwise in Fish. If there's a workaround to get this working, please do let me know as I'd prefer using Oh-My-Posh in Fish for consistency.

### Theme

Init/Custom DarkBlood

### What OS are you seeing the problem on?

Windows

### Which shell are you using?

fish

### Log output

```Shell
xXChr@LEGION /c> oh-my-posh debug

Version: 11.3.1

Segments:

ConsoleTitle(false) -   0 ms -
shell(true)    -   0 ms - ┏[fish]
session(true)  -   1 ms - [xXChr]
git(false)     -   8 ms -
root(false)    -   0 ms -
path(true)     -   0 ms - ┖[C:\]>

Run duration: 50.5539ms

Cache path: C:\Users\xXChr\AppData\Local\oh-my-posh

Config path: C:\Users\xXChr\.onedarkblood.omp.json

Logs:

2022/10/10 05:26:16 debug: Getenv
C:\Users\xXChr\AppData\Local
2022/10/10 05:26:16 Getenv duration: 2.4336ms, args: LOCALAPPDATA
2022/10/10 05:26:16 CachePath duration: 2.9509ms, args:
2022/10/10 05:26:16 debug: Getenv
C:\Users\xXChr\.onedarkblood.omp.json
2022/10/10 05:26:16 Getenv duration: 0s, args: POSH_THEME
2022/10/10 05:26:16 debug: Shell
process name: fish.exe
2022/10/10 05:26:16 Shell duration: 15.0479ms, args:
2022/10/10 05:26:16 resolveConfigPath duration: 15.0479ms, args:
2022/10/10 05:26:16 Init duration: 39.4188ms, args:
2022/10/10 05:26:16 Flags duration: 0s, args:
2022/10/10 05:26:16 config.loadConfig duration: 1ms, args:
2022/10/10 05:26:16 Flags duration: 0s, args:
2022/10/10 05:26:16 debug: Getenv

2022/10/10 05:26:16 Getenv duration: 0s, args: OMP_CACHE_DISABLED
2022/10/10 05:26:16 WindowsRegistryKeyValue duration: 0s, args: HKEY_CURRENT_USER\Software\Microsoft\Windows\DWM\ColorizationColor
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 debug: WindowsRegistryKeyValue
ColorizationColor(DWORD): 0xC4E2241A
2022/10/10 05:26:16 Root duration: 0s, args:
2022/10/10 05:26:16 Shell duration: 0s, args:
2022/10/10 05:26:16 ErrorCode duration: 0s, args:
2022/10/10 05:26:16 IsWsl duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 debug: Pwd
C:\
2022/10/10 05:26:16 Pwd duration: 0s, args:
2022/10/10 05:26:16 debug: Home
C:\msys64\home\xXChr
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 debug: User
xXChr
2022/10/10 05:26:16 User duration: 0s, args:
2022/10/10 05:26:16 debug: Host
LEGION
2022/10/10 05:26:16 Host duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 TemplateCache duration: 0s, args:
2022/10/10 05:26:16 Flags duration: 0s, args:
2022/10/10 05:26:16 debug: Pwd
C:\
2022/10/10 05:26:16 Pwd duration: 0s, args:
2022/10/10 05:26:16 debug: Home
C:\msys64\home\xXChr
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 Shell duration: 0s, args:
2022/10/10 05:26:16 Flags duration: 0s, args:
2022/10/10 05:26:16 TemplateCache duration: 0s, args:
2022/10/10 05:26:16 TemplateCache duration: 0s, args:
2022/10/10 05:26:16 Shell duration: 0s, args:
2022/10/10 05:26:16 debug: Pwd
C:\
2022/10/10 05:26:16 Pwd duration: 0s, args:
2022/10/10 05:26:16 debug: Home
C:\msys64\home\xXChr
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 debug: Getenv

2022/10/10 05:26:16 Getenv duration: 0s, args: SSH_CONNECTION
2022/10/10 05:26:16 debug: Getenv

2022/10/10 05:26:16 Getenv duration: 0s, args: SSH_CLIENT
2022/10/10 05:26:16 TemplateCache duration: 0s, args:
2022/10/10 05:26:16 TemplateCache duration: 0s, args:
2022/10/10 05:26:16 Shell duration: 0s, args:
2022/10/10 05:26:16 debug: Pwd
C:\
2022/10/10 05:26:16 Pwd duration: 0s, args:
2022/10/10 05:26:16 debug: Home
C:\msys64\home\xXChr
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 debug: CommandPath
C:\Program Files\Git\cmd\git.exe
2022/10/10 05:26:16 CommandPath duration: 8.9398ms, args: git.exe
2022/10/10 05:26:16 HasCommand duration: 8.9398ms, args: git.exe
2022/10/10 05:26:16 debug: Pwd
C:\
2022/10/10 05:26:16 Pwd duration: 0s, args:
2022/10/10 05:26:16 error: HasParentFilePath
CreateFile C:\.git: The system cannot find the file specified.
2022/10/10 05:26:16 HasParentFilePath duration: 0s, args: .git
2022/10/10 05:26:16 debug: Pwd
C:\
2022/10/10 05:26:16 Pwd duration: 0s, args:
2022/10/10 05:26:16 debug: Home
C:\msys64\home\xXChr
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 Root duration: 0s, args:
2022/10/10 05:26:16 Flags duration: 0s, args:
2022/10/10 05:26:16 debug: Pwd
C:\
2022/10/10 05:26:16 Pwd duration: 0s, args:
2022/10/10 05:26:16 debug: Home
C:\msys64\home\xXChr
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 Shell duration: 0s, args:
2022/10/10 05:26:16 Shell duration: 0s, args:
2022/10/10 05:26:16 debug: Pwd
C:\
2022/10/10 05:26:16 Pwd duration: 0s, args:
2022/10/10 05:26:16 debug: Home
C:\msys64\home\xXChr
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 PathSeparator duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 PathSeparator duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 PathSeparator duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 PathSeparator duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 PathSeparator duration: 0s, args:
2022/10/10 05:26:16 GOOS duration: 0s, args:
2022/10/10 05:26:16 PathSeparator duration: 0s, args:
2022/10/10 05:26:16 IsWsl duration: 0s, args:
2022/10/10 05:26:16 StackCount duration: 0s, args:
2022/10/10 05:26:16 DirIsWritable duration: 0s, args:
2022/10/10 05:26:16 TemplateCache duration: 0s, args:
2022/10/10 05:26:16 TemplateCache duration: 0s, args:
2022/10/10 05:26:16 Shell duration: 0s, args:
2022/10/10 05:26:16 debug: Getenv
C:\Users\xXChr\AppData\Local
2022/10/10 05:26:16 Getenv duration: 0s, args: LOCALAPPDATA
2022/10/10 05:26:16 CachePath duration: 0s, args:
2022/10/10 05:26:16 Flags duration: 0s, args:
```

**Comment by JanDeDobbeleer** [comment]
@Terraphice it can't resolve the path here, as it's MSYS. Initialise using the `--strict` parameter:

```
oh-my-posh init fish --strict | source
```

That said, if you want to use fish, I suggest to use WSL2. I would love people to stop using MSYS as it's the worst environment ever to maintain (for me).

**Comment by Terraphice** [comment]
@JanDeDobbeleer I knew I saw a fix about something like this somewhere in the docs, but it's late and after ages of searching I gave up. I really do appreciate your help.

I know, I know. MSYS2 is a real ####. I use WSL2 a bit, and my actual daily driver is Arch btw. I would not even consider using MSYS if it wasn't for a very specific purpose I have at the moment.

(Long story short, I've been bet that I can't entirely, fully rice Windows and get a large portion of Linux tools (or equivalents) working within TUI/CLI. My journey over the past week has lead to me corrupting half of the Registry, the entirety of the MS store, rolling back the OS, etc. multiple times. If anything, I'm doing it for the challenge and to see what I can get away with on Windows.)

Again, thanks a ton for the quick solution. (Obviously it worked. I'm a bozo.) 👍
